src/NuGet.Clients/PackageManagement.VisualStudio/ProjectSystems/CpsPackageReferenceProject.cs

```diff
@@ -172,21 +177,49 @@ private static PackageReference ToPackageReference(LibraryDependency library, Nu
             return new PackageReference(identity, targetFramework);
         }
 
-        public override async Task<Boolean> InstallPackageAsync(PackageIdentity packageIdentity, DownloadResourceResult downloadResourceResult, INuGetProjectContext nuGetProjectContext, CancellationToken token)
+        public override async Task<Boolean> InstallPackageAsync(PackageIdentity packageIdentity,
+            INuGetProjectContext nuGetProjectContext,
+            IEnumerable<NuGetFramework> successfulFrameworks,
+            IEnumerable<NuGetFramework> unsuccessfulFrameworks,
+            CancellationToken token)
         {
-
             nuGetProjectContext.Log(MessageLevel.Info, Strings.InstallingPackage, packageIdentity);
 
-            var configuredProject = await _unconfiguredProject.GetSuggestedConfiguredProjectAsync();
-            var result = await
-                configuredProject?.Services.PackageReferences.AddAsync
-                (packageIdentity.Id, packageIdentity.Version.ToNormalizedString());
-            
-            // This is the update operation
-            if (result != null && !result.Added)
+            if (successfulFrameworks.Any() && unsuccessfulFrameworks.Any())
+            {
+                var conditionalService = _unconfiguredProject
+                    .Services
+                    .ExportProvider
+                    .GetExportedValue<IConditionalPackageReferencesService>();
+
+                if (conditionalService == null)
+                {
+                    return true;
```

Comment:
I see a similar behavior in the unconditional case when `configuredProject` is null or if the `result` is null (both exceptional cases AFAIK). I had assumed this was by design. Alternatives include logging or throwing. Preference?